src/components/Calculator.js

```diff
@@ -1,44 +1,62 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = { };
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((state) => ({
+      ...state,
+      ...calculate(state, e.target.value),
+    }));
   }
 
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div className="calculator-container d-flex flex-d-c">
-        <div className="calculator-output d-flex j-content-flex-end">0</div>
+        <div className="calculator-output d-flex j-content-flex-end">
+          {total}
+          {operation}
+          {next}
+        </div>
         <div className="d-flex flex-d-c">
           <div className="d-flex">
-            <button type="button" className="operator operatorTop">AC</button>
-            <button type="button" className="operator operatorTop">+/-</button>
-            <button type="button" className="operator operatorTop">%</button>
-            <button type="button" className="operator operatorTop operatorRight  divide">➗</button>
+            <button type="button" className="operator operatorTop" value="AC" onClick={this.handleClick}>AC</button>
+            <button type="button" className="operator operatorTop" value="+/-" onClick={this.handleClick}>+/-</button>
+            <button type="button" className="operator operatorTop" value="%" onClick={this.handleClick}>%</button>
+            <button type="button" className="operator operatorTop operatorRight" value="÷" onClick={this.handleClick}>÷</button>
           </div>
           <div className="d-flex">
-            <button type="button" className="operand" data="7">7</button>
-            <button type="button" className="operand" data="8">8</button>
-            <button type="button" className="operand" data="9">9</button>
-            <button type="button" className="operator operatorRight" data="x">x</button>
+            <button type="button" className="operand" value="7" onClick={this.handleClick}>7</button>
+            <button type="button" className="operand" value="8" onClick={this.handleClick}>8</button>
+            <button type="button" className="operand" value="9">9</button>
```

Comment:
- [ ] Kindly make sure to add click event for number 9, perhaps you forgot to write it. This button is not working because of it.

Reply:
Hi @sja-thedude 
Thanks for your kind and clear feedback, will right away work on it.